Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityToolkitFiles.cs

```diff
@@ -246,5 +206,70 @@ public static string MapRelativeFilePath(MixedRealityToolkitModuleType module, s
             }
             return null;
         }
+
+        private static readonly Dictionary<string, MixedRealityToolkitModuleType> moduleNameMap = new Dictionary<string, MixedRealityToolkitModuleType>()
+        {
+            { "", MixedRealityToolkitModuleType.Core },
+            { "Generated", MixedRealityToolkitModuleType.Generated },
+            { "Providers", MixedRealityToolkitModuleType.Providers },
+            { "Services", MixedRealityToolkitModuleType.Services },
+            { "SDK", MixedRealityToolkitModuleType.SDK },
+            { "Examples", MixedRealityToolkitModuleType.Examples },
+            { "Tests", MixedRealityToolkitModuleType.Tests }
+        };
+
+        private static bool FindMatchingModule(string folder, out MixedRealityToolkitModuleType result)
```

Comment:
Can we have a few basic unit tests for this function? i.e. something like:

test 1) pass in a unity package version that's at the root.
test 2) pass in a unity package version that's in the Packages/ folder
test 3) pass in a unity package version that's in some other arbitrary location
test 4) pass in the nuget version
test 5) totally unrelated path that shouldn't find a match.

Really this would also help to canon-ize (is that a word?) the types of paths we expect and show how this handles them

Reply:
Writing exhaustive tests for this utility class is a bit tricky, at least in its current state. It's a static class that searches all the folders at construction time, i.e. can't really test it on a clean slate with no existing folders. Can still test adding/removing folders in the existing project, although sandboxing changes in the file system for tests may require some work.

I'll try to get something started at least.